core/src/impl/Kokkos_FunctorAnalysis.hpp

```diff
@@ -90,6 +90,29 @@ struct DeduceFunctorPatternInterface<ParallelScanWithTotal<
   using type = FunctorPatternInterface::SCAN;
 };
 
+#if defined(__CUDA_ARCH__)
```

Comment:
Would you please use KOKKOS_IF_ON_HOST/DEVICE and guard if CUDA is enabled?

Reply:
We end up with unused arguments that produce compiler warnings doing it that way. Once we have c++17 support enabled, I could mark them with `[[maybe_unused]]`, but that's not an option today.

This may all be moot if the other comments, about the fixed synchronization obviating volatile, holds up.

Reply:
In that case, I could delete this function entirely

Reply:
Testing that in #4931